### ISSUE TITLE
Nested variables don't refresh

### ISSUE THREAD

**Issue by bode689** [marcusolsson/grafana-json-datasource#216]
Hello,

Within my Grafana application (v 8.2.3), it is very useful to nest variables from left to right, in order to let the user filter efficiently.

I've been doing this for a while with InfluxDB datasource without any problem.
Now, I'm integrating other sources using this great JSON API plugin but I'm realizing that nested query variables never update, unless you refresh the entire dashboard.
For example I'd like to use the returned value of variable A as a parameter value of query variable B.
The same query works perfectly if used in a panel and not in a variable.

I haven't found any open issue on this, please correct me if I'm wrong!

Is there any solution or trick to sort this out?

Thank you in advance

**Comment by ugoleleux** [comment]
Hi,

Same issue for me on Grafana 8.2.6 and 8.3.0

Ugo

**Comment by marcusolsson** [comment]
This looks like a duplicate of https://github.com/marcusolsson/grafana-json-datasource/issues/95.

I haven't been able to figure this out unfortunately. It seems like the plugin doesn't have any control of this (but obviously this works for others).

**Comment by ugoleleux** [comment]
Hi,
Thank you for your reply.

The case I opened on the grafana repo has been closed as the problem does not occur with other plugins...
This is a dead end for me

**Comment by marcusolsson** [comment]
Most likely this is a bug with this plugin. I'm closing this as a duplicate. If you come across any more details that can help, please comment in #95.

**Comment by ugoleleux** [comment]
@marcusolsson 
I found a new behavior that will help you in your analysis.
I have a variable $name that contains a list of choices.
This variable is used inside a child variable on the Params tab :
![image](https://user-images.githubusercontent.com/95695567/145830650-633211e9-85e5-433a-9236-b11f21044859.png)
When I look at the dependency tree, the $name var is not used by $child

If I refer $name variable in the Path tab :
![image](https://user-images.githubusercontent.com/95695567/145830939-9c5c1227-f3d9-4616-818f-f8da39cd8cb0.png)
This time, the dependency tree is well seeing the relation.

By doing this little change, my child var is now updated on the fly when I modify the $name.

I hope this will help you to correct this issue :)

Regards
Ugo
